app/src/main/java/com/concordia/mcga/adapters/BuildingInformationArrayAdapter.java

```diff
@@ -0,0 +1,212 @@
+package com.concordia.mcga.adapters;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.concordia.mcga.activities.R;
+
+import java.util.ArrayList;
+
+import static android.content.ContentValues.TAG;
+import static com.concordia.mcga.activities.R.id.imageView;
+
+
+public class BuildingInformationArrayAdapter extends ArrayAdapter<String[]> {
+
+    ////////////////////////////////////////////////////////////
+    // INSTANCE VARIABLES
+    ////////////////////////////////////////////////////////////
+
+    private final int IMAGES_PER_ROW = 4;
+
+    private final Context context;
+    private ArrayList<String[]> rowImages = new ArrayList<String[]>();
```

Comment:
I know that this is how it was implemented, but it could also have been implemented with a nested List inside a List or a 2-dimensional array, and this is exactly what I'm suggesting. If we leave it the way it is just because it's already there, then we are left to use a confusing and clunky data structure.